File: routes/ticketRoutes.js
// API Routes
const express = require("express");
const router = express.Router();
const ticketController = require('../controllers/ticketController')
router.post('/tickets', ticketController.createTicket); // Create a ticket
router.get('/tickets', ticketController.getTickets); // Get all tickets
router.get('/tickets/:id', ticketController.getTicket); // Get a single ticket by ID
router.put('/tickets/:id', ticketController.updateTicket); // Update a ticket
router.delete('/tickets/:id', ticketController.deleteTicket); // Delete a ticket

module.exports = router;

File: routes/ticket_routes.py
from fastapi import APIRouter
from controllers.ticket_controller import TicketController

router = APIRouter(prefix="/tickets", tags=["Tickets"])

@router.post("", response_model=dict)
async def create_ticket(data: dict):
    return await TicketController.create_ticket(data)

@router.get("", response_model=list)
async def get_all_tickets():
    return await TicketController.get_all_tickets()

@router.get("/{id}", response_model=dict)
async def get_ticket_by_id(id: str):
    return await TicketController.get_ticket_by_id(id)

@router.put("/{id}", response_model=dict)
async def update_ticket(id: str, data: dict):
    return await TicketController.update_ticket(id, data)

@router.delete("/{id}", response_model=dict)
async def delete_ticket(id: str):
    return await TicketController.delete_ticket(id)